mungegithub/github/github.go

```diff
@@ -738,7 +709,42 @@ func (config *Config) SetBranchProtection(name string, contexts []string) error
 		return nil
 	}
 
-	return config.setBranchProtection(name, contexts)
+	request := &github.ProtectionRequest{
+		RequiredStatusChecks: &github.RequiredStatusChecks{
+			Strict:   false,
+			Contexts: contexts,
+		},
+		RequiredPullRequestReviews: prot.RequiredPullRequestReviews,
+		Restrictions:               unchangedRestrictionRequest(prot.Restrictions),
+		EnforceAdmins:              false,
+	}
+	return config.setBranchProtection(name, request)
+}
+
+// unchangedRestrictionRequest generates a request that will
+// not make any changes to the teams and users that can merge
+// into a branch
+func unchangedRestrictionRequest(restrictions *github.BranchRestrictions) *github.BranchRestrictionsRequest {
+	request := &github.BranchRestrictionsRequest{
+		Users: []string{},
+		Teams: []string{},
+	}
+
+	if restrictions == nil {
+		return request
```

Comment:
Should nil be returned here instead of an empty request?

Reply:
Should be same functionality, but yes we can return `nil`